stock_return_request/models/stock_return_request.py

```diff
@@ -0,0 +1,517 @@
+    )
+    state = fields.Selection(
+        selection=[
+            ('draft', 'Open'),
+            ('confirmed', 'Confirmed'),
+            ('done', 'Done'),
+            ('cancel', 'Cancelled'),
+        ],
+        default='draft',
+        readonly=True,
+        copy=False,
+        track_visibility='onchange',
+    )
+    returned_picking_ids = fields.One2many(
+        comodel_name='stock.picking',
+        inverse_name='stock_return_request_id',
+        string='Returned Pickings',
+        readonly=True,
+        copy=False,
+    )
+    to_refund = fields.Boolean(
+        string="To refund",
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    show_to_refund = fields.Boolean(
+        compute='_compute_show_to_refund',
+        default=lambda self: 'to_refund' in self.env['stock.move']._fields,
+        readonly=True,
+        help="Whether to show it or not depending on the availability of"
+             "the stock_account module (so a bridge module is not necessary)",
+    )
+    line_ids = fields.One2many(
+        comodel_name='stock.return.request.line',
+        inverse_name='request_id',
+        string='Stock Return',
+        copy=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    note = fields.Text(
+        string="Comments",
+        help="They will be visible on the report",
+    )
+
+    @api.onchange('return_type', 'partner_id')
+    def onchange_locations(self):
+        """UI helpers to determine locations"""
+        warehouse = self._default_warehouse_id()
+        if self.return_type == 'supplier':
+            self.return_to_location = (
+                self.partner_id.property_stock_supplier)
+            if self.return_from_location.usage != 'internal':
+                self.return_from_location = warehouse.lot_stock_id.id
+        if self.return_type == 'customer':
+            self.return_from_location = (
+                self.partner_id.property_stock_customer)
+            if self.return_to_location.usage != 'internal':
+                self.return_to_location = warehouse.lot_stock_id.id
+        if self.return_type == 'internal':
+            self.partner_id = False
+            if self.return_to_location.usage != 'internal':
+                self.return_to_location = warehouse.lot_stock_id.id
+            if self.return_from_location.usage != 'internal':
+                self.return_from_location = warehouse.lot_stock_id.id
+
+    @api.model
+    def _default_warehouse_id(self):
+        warehouse = self.env['stock.warehouse'].search([
+            ('company_id', '=', self.env.user.company_id.id),
+        ], limit=1)
+        return warehouse
+
+    def _compute_show_to_refund(self):
+        for one in self:
+            if 'to_refund' not in self.env['stock.move']._fields:
+                continue
+            one.show_to_refund = True
+
+    def _create_picking(self, picking, picking_moves):
```

Comment:
Can you do this to pass extra vals avoiding extra write calls:
```suggestion
    def _create_picking(self, picking, picking_moves, extra_vals):
```

Reply:
Better option is create _prepare_picking and use copy_data function:
https://github.com/odoo/odoo/blob/dd620c788fe46131db161449f1f988083d700aac/odoo/models.py#L3848

Reply:
Sure! I went for the second choice. Thanks!